WordPress/src/main/java/org/wordpress/android/ui/uploads/MediaUploadHandler.java

```diff
@@ -251,14 +262,27 @@ private void cancelUpload(MediaModel oneUpload, boolean delete) {
         }
     }
 
-    private void dispatchUploadAction(@NonNull final MediaModel media, @NonNull final SiteModel site) {
+    private void dispatchUploadAction(@NonNull final MediaModel media, boolean optimize) {
```

Comment:
can we refactor this a bit? i.e. the `optimize` code path only makes sense for videos, and `dispatching` and `optimizing` are two quite different things. I'd go with a wrapping method maybe called `prepareForUpload` that internally decides to `dispatchUploadAction` directly or call `new VideoOptimizer(media, this).start()` if this is a video and needs to be optimized.